arangod/Cluster/ClusterInfo.cpp

```diff
@@ -1600,7 +1611,8 @@ int ClusterInfo::createCollectionCoordinator(std::string const& databaseName,
   auto dbServerResult = std::make_shared<int>(-1);
```

Comment:
This probably has to be made atomic as well.

Reply:
I am unsure, there's some weird locking going on here

Reply:
Yes, the function is very long and confusing (IMHO). Are you referring to the `cacheMutex`? Because the access to `dbServerResult` in line 1812 seems to be outside the scope of the `cacheMutex` locker in line 1740